src/machine/asm/mod.rs

```diff
@@ -432,8 +473,11 @@ impl Memory for Box<AsmCoreMachine> {
             check_memory(self, page);
             self.set_flag(page, FLAG_DIRTY)?;
         }
-        let slice = &mut self.memory[addr as usize..addr as usize + value.len()];
-        slice.copy_from_slice(value);
+        unsafe {
+            let memory = self.memory_ptr as *mut u8;
```

Comment:
I'm seeing this pattern everywhere, can we add a function to AsmCoreMachine where we cast the raw pointer to a proper slice? That would eliminate most unsafe usage here.

Reply:
Of course.